tests/lib/rules/no-wait-for-side-effects.test.ts

```diff
@@ -797,5 +807,19 @@ ruleTester.run(RULE_NAME, rule, {
 				],
 			} as const,
 		]),
+
+		...SUPPORTED_TESTING_FRAMEWORKS.flatMap((testingFramework) => [
```

Comment:
suggestion (blocking): Please add a test for the scenario reported in #742 description.

Reply:
Done